### ISSUE TITLE
[PATCH 5.6.0-RC2] hints/freebsd.sh - make perl work on FreeBSD 5.0

### ISSUE THREAD

**Issue by p5pRT** [Perl/perl5#1542]
Migrated from [rt.perl.org#2759](https://rt-archive.perl.org/perl5/Ticket/Display.html?id=2759) (status was 'resolved')

Searchable as RT2759$

**Comment by p5pRT** [comment]
### From @tobez

Without this patch\, 5\.6\.0\-RC2 does not compile on FreeBSD 5\.0\-current
\(which has just been born\)\.  With it\, all tests are successful for both

\-des and \-Dusethreads \-des configurations\.

Cheers\, Anton\.

<details><summary>Inline Patch</summary>

```diff
--- freebsd.sh.orig	Wed Mar 15 14:43:24 2000
+++ freebsd.sh	Wed Mar 15 14:30:55 2000
@@ -99,7 +99,7 @@
 case "$osvers" in
 0.*|1.0*) ;;
 
-3.*|4.0*)
+3.*|4.*|5.0*)
         objformat=`/usr/bin/objformat`
         if [ x$objformat = xelf ]; then
             libpth="/usr/lib /usr/local/lib"
@@ -122,8 +122,8 @@
 esac
 
 case "$osvers" in
-4.0*)
-	if /usr/bin/file /usr/lib/libc.so.3 | /usr/bin/grep -vq "not stripped" ; then
+4.*|5.0*)
+	if /usr/bin/file -L /usr/lib/libc.so | /usr/bin/grep -vq "not stripped" ; then
 	    usenm=false
 	fi
 esac
@@ -173,7 +173,7 @@
 $define|true|[yY]*)
         lc_r=`/sbin/ldconfig -r|grep ':-lc_r'|awk '{print $NF}'|tail -1`
         case "$osvers" in  
-	2.2.8*|3.*|4.*)
+	2.2.8*|3.*|4.*|5.0*)
 	      if [ ! -r "$lc_r" ]; then
 	      cat <<EOM >&4
 POSIX threads should be supported by FreeBSD $osvers --
```
</details>

**Comment by p5pRT** [comment]
### From @doughera88

On Wed\, 15 Mar 2000\, Anton Berezin wrote&#8203;:

> Without this patch\, 5\.6\.0\-RC2 does not compile on FreeBSD 5\.0\-current
> \(which has just been born\)\.  With it\, all tests are successful for both
> 
> \-3\.\*|4\.0\*\)
> \+3\.\*|4\.\*|5\.0\*\)

But that will not be useful once 5\.1 comes out\, which could certainly
happen during perl\-5\.6\.0's lifetime\.

Quoting hints/README\.txt&#8203;:

&nbsp;    If you need to test for specific minor version differences in your
&nbsp;    hints file\, be sure to include a default choice\.  \(See aix\.sh for one
&nbsp;    example\.\) That way\, if you write a hint file for foonix 3\.2\, it might
&nbsp;    still work without any changes when foonix 3\.3 is released\.

That is\, do something like

case "$osvers" in
\[0\-2\]\*\) \# do nothing
&nbsp;    ;;
\*\)    \# Known to work at least from 3\.0 \- 5\.0
&nbsp;    \# do whatever needs to be done\.
&nbsp;    ;;
esac


&nbsp;    Andy Dougherty        doughera@&#8203;lafayette\.edu
&nbsp;    Dept\. of Physics
&nbsp;    Lafayette College\, Easton PA 18042

**Comment by p5pRT** [comment]
### From @tobez

On Wed\, Mar 15\, 2000 at 11&#8203;:40&#8203;:17AM \-0500\, Andy Dougherty wrote&#8203;:
> On Wed\, 15 Mar 2000\, Anton Berezin wrote&#8203;:
> 
> > Without this patch\, 5\.6\.0\-RC2 does not compile on FreeBSD 5\.0\-current
> > \(which has just been born\)\.  With it\, all tests are successful for both
> > 
> > \-3\.\*|4\.0\*\)
> > \+3\.\*|4\.\*|5\.0\*\)
> 
> But that will not be useful once 5\.1 comes out\, which could certainly
> happen during perl\-5\.6\.0's lifetime\.

Nope \- it will not happen until at least April 2001\.

Last time when I submitted similar thing for 4\.\*\, I was corrected\, and
4\.0\* was put in freebsd\.sh instead\.  :\-\)

Cheers\,
\-\- 
Anton Berezin \<tobez@&#8203;plab\.ku\.dk>
The Protein Laboratory\, University of Copenhagen

**Comment by p5pRT** [comment]
### From @doughera88

On Wed\, 15 Mar 2000\, Anton Berezin wrote&#8203;:

> Last time when I submitted similar thing for 4\.\*\, I was corrected\, and
> 4\.0\* was put in freebsd\.sh instead\.  :\-\)

Then whoever did that "correcting" was wrong :\-\)\.

&nbsp;    Andy Dougherty        doughera@&#8203;lafayette\.edu

**Comment by p5pRT** [comment]
### From @tobez

On Wed\, Mar 15\, 2000 at 11&#8203;:58&#8203;:57AM \-0500\, Andy Dougherty wrote&#8203;:

>> Last time when I submitted similar thing for 4\.\*\, I was corrected\,
>> and 4\.0\* was put in freebsd\.sh instead\.  :\-\)

> Then whoever did that "correcting" was wrong :\-\)\.

Well\, I am not so sure about that\.  The problem is\, we are not talking
here about \`specific minor version differences';  rather\, 5\.0\-current as
of March 2000\, and 5\.0\-current as of March 2001 \(and 5\.1\-release as of
April 2001\) will be very different\.  5\.0 just forked off 4\.0\, and has a
long time ahead\, with lots of changes coming\.

So basically\, I think that 5\.0\* is as reasonable \_today\_ as 5\.\* is \-
there is a good chance of additional hits tweaking to be done later this
year\.

Cheers\,
\-\- 
Anton Berezin \<tobez@&#8203;plab\.ku\.dk>
The Protein Laboratory\, University of Copenhagen

**Comment by p5pRT** [comment]
### From @doughera88

On Wed\, 15 Mar 2000\, Anton Berezin wrote&#8203;:

> On Wed\, Mar 15\, 2000 at 11&#8203;:58&#8203;:57AM \-0500\, Andy Dougherty wrote&#8203;:
> 
> >> Last time when I submitted similar thing for 4\.\*\, I was corrected\,
> >> and 4\.0\* was put in freebsd\.sh instead\.  :\-\)
> 
> > Then whoever did that "correcting" was wrong :\-\)\.
> 
> Well\, I am not so sure about that\.

I am :\-\)\.

>                                     The problem is\, we are not talking
> here about \`specific minor version differences';  rather\, 5\.0\-current as
> of March 2000\, and 5\.0\-current as of March 2001 \(and 5\.1\-release as of
> April 2001\) will be very different\.  5\.0 just forked off 4\.0\, and has a
> long time ahead\, with lots of changes coming\.

Sure\, but not necessarily all of them will require changes the hints file\.
As a concrete example\, in some cases your patch just changed 4\.0\* into
4\.0\*|5\.0\*\.  Had the previous version of the hints file just included the
most recent case as the default\, 5\.0 would have required NO CHANGES in
those parts of the hints file\.

As a second concrete example\, consider perl5\.005\_03's hints/freebsd\.sh\.
5\.005\_03 is still the official maintenance version\, but it doesn't
hand FreeBSD 5\.0 gracefully at all\.

You have to remember that hints files need to have a long shelf life\. The
current maintenance version 5\.005\_03\, has been around almost a full year\.  
Assume the worst\.  Assume that the hints file you write today is going to
be used \_exactly as is\_ for at least the next year\, possibly longer\.  
Thus assume that folks will try to use perl\-5\.6\.0 with FreeBSD 5\.1\.

> So basically\, I think that 5\.0\* is as reasonable \_today\_ as 5\.\* is \-
> there is a good chance of additional hints tweaking to be done later
> this year\.

But you'll only have a chance to do that tweaking if there are more
maintenance releases of perl during the year as things evolve\.  That's not
guaranteed\.  You can't assume you'll have that chance\.

I would use neither 5\.0\* nor 5\.\*\.  I would simply use \*\.  Sure\, it's quite
possible that when someone tries to run this hints file a year from now
under 5\.1 that the hints won't be exactly right\.  However\, the 5\.0\-based
guess is probably more likely to be right than no guess at all\!

Sorry to go on so long about this\.  It really isn't that big a deal\, but
it's also not hard to program defensively in this instance\.  I can't see
any reason not to\.

&nbsp;    Andy Dougherty        doughera@&#8203;lafayette\.edu
&nbsp;    Dept\. of Physics
&nbsp;    Lafayette College\, Easton PA 18042

**Comment by p5pRT** [comment]
### From @tobez

On Wed\, Mar 15\, 2000 at 12&#8203;:39&#8203;:23PM \-0500\, Andy Dougherty wrote&#8203;:

> Assume the worst\.  Assume that the hints file you write today is going
> to be used \_exactly as is\_ for at least the next year\, possibly
> longer\.  Thus assume that folks will try to use perl\-5\.6\.0 with
> FreeBSD 5\.1\.

> I would use neither 5\.0\* nor 5\.\*\.  I would simply use \*\.  Sure\, it's
> quite possible that when someone tries to run this hints file a year
> from now under 5\.1 that the hints won't be exactly right\.  However\,
> the 5\.0\-based guess is probably more likely to be right than no guess
> at all\!

> Sorry to go on so long about this\.  It really isn't that big a deal\,
> but it's also not hard to program defensively in this instance\.  I
> can't see any reason not to\.

Thanks\.  With this explanation\, I am totally convinced\.  Expect reworked
patch as soon as I finish its testing on 3\.4\, 4\.0\, and 5\.0\.

Cheers\,
\-\- 
Anton Berezin \<tobez@&#8203;plab\.ku\.dk>
The Protein Laboratory\, University of Copenhagen
